src/fs_io.cc

```diff
@@ -37,6 +37,23 @@ static void cxx_xfree(void *ptr)
     xfree(ptr);
 }
 
+dwrite_q::dwrite_q(const size_t aSize, char *aBuffer, FREE *aFree) :
+    buf(aBuffer),
+    len(aSize),
+    free_func(aFree)
+{
+    if (!buf) {
+        buf = static_cast<char *>(xmalloc(len));
```

Comment:
Is there a reason we can't use memAllocBuf here?

Reply:
That is one of the things I would like to do as followup.
For this PR the scope is only changing the Memory Pool used by the ` class dwrite_q` object itself.